acl_linux.go

```diff
@@ -183,6 +201,41 @@ func getType(path, attr string) (ACL, error) {
 	}
 }
 
+func fgetType(f *os.File, attr string) (ACL, error) {
```

Comment:
Can you deduplicate this and `getType`, and also `setType` and `fsetType`? My suggestion would be to add an interface that abstracts over the difference between a path and an `os.File`, something like:

```go
type file interface {
    Getxattr(attr string, dest []byte) (sz int, err error)
    Setxattr(attr string, dest []byte, flags int) (err error)
    Stat() (FileInfo, error)
}
```

Then you could have a single implementation of `getType` and a single implementation of `setType` that operate on `file`s rather than `string`s.